src/components/SettingsDialog/SettingsDialog.tsx

```diff
@@ -19,14 +19,18 @@ import {ReactComponent as AppearanceIcon} from "assets/icon-appearance.svg";
 import {ReactComponent as ExportIcon} from "assets/icon-export.svg";
 import {ReactComponent as FeedbackIcon} from "assets/icon-feedback.svg";
 import "./SettingsDialog.scss";
+import {ConfirmationDialog} from "components/ConfirmationDialog";
+import {Actions} from "store/action";
 
 export const SettingsDialog: FC = () => {
   const {t} = useTranslation();
   const navigate = useNavigate();
   const boardId = useAppSelector((applicationState) => applicationState.board.data!.id);
-  const me = useAppSelector((applicationState) => applicationState.participants?.self.user);
+  const me = useAppSelector((applicationState) => applicationState.participants!.self.user);
   const isBoardModerator = useAppSelector((state) => state.participants?.self.role === "MODERATOR" || state.participants?.self.role === "OWNER");
   const feedbackEnabled = useAppSelector((state) => state.view.feedbackEnabled);
+  const unsavedAvatar = useAppSelector((state) => state.participants?.self.user.unsavedAvatar);
```

Comment:
In my opinion, this shouldn't be saved in our global redux store. I think it would be much better to just have that as a component state because this only concerns the SettingsDialog and nothing else. Maybe you could work with callbacks or contexts instead. 

I actually think it would be a great idea to implement a context for the settings dialog (and its children).

Reply:
Thank you for this good advice. 
I've changed the implementation to use a settings context instead of redux.